### ISSUE TITLE
Smooth Scrolling: Difference Between VS Code / Chrome

### ISSUE THREAD

**Issue by hediet** [microsoft/vscode#185891]
* [Monaco Editor Scrolling](https://microsoft.github.io/monaco-editor/playground.html?source=v0.40.0-dev.20230621#XQAAAAJUFgAAAAAAAABBqQkHQ5NjdM0b2-jY7SIQ9S7DNlzs5W-mwj0fe1ZCDRFc9ws9XQE0SJE1jc2Vhu4mzIDyzq0Mcjp2sHIoP0Q_KCcJzV8B2IG4fZn4lWB1k8nG_wxyB-Maregzpyoqu8LykKh5oDCZw4ZZPyp7lquRonagZ1aKuVWkIIWPIuFXL9XeNyBlBZ8UrRrroHR4xt8uHJ44Q4QpjNYAOoEvPDvRNO8yV-YG7549C6WC6r8nNhD75FQS1yzKMpZuist1JFmXQ5rtD6Ndj2YBOMmVSD_ijeHstrBIt9PMAr0rxlrHivjkusmHDaJvWE0SQUkZKCZpEeNPatXn03hEMoHYS4y-OXAJXcbwRga9VaQrKuW-A4NvyIdgpZeAj-XU93ud7P2IXHYbUhU7eVaygey3igUqvsYOGBb-COciovDAza8Uc4jFxQCN2K6IQAjOOF2Zb739Ad1mOss_VS_-JRkRyqKfSgiApPCB-KB7Vj6xqRutkXfKUt4ZBM6P6CdqEYrd2qTR0TNkQoovTVJ7tHK8jY09FdErBAmbPokwUQ5Kd8fw6FZk1aWkQ9Ct6tghg1DfBUGXZelG2wM7LWFj3kyWeptScz01B0zUE56qVNpARvn8ZQ3f4c0OhP_NrKuDBOm1bZ6r8JlvfHYZhlvVv2HjOpEj47IDfaCoRbKozELbV080fm8grRE1RPG0FKfKcXiJ_fgQAQBRZQ4uZsDdy0qPcl_nUOyUtDijo0nDJq9GKKQJfbqL61yxAnPnlnYdsciu6OE_U-43wU3eYIfMpZeuk35bt5y7mayimqOAvRHiVJ_dA4Z5o7SVsqM6UO0vxwqktDsSnXfjB2-l4yfvO6GROmGzm_2ROouRZc4GWzK_azV44b9R_al_985RhePERlGK2z9sWuThUg-pqnMJPXIfCL2CYXOGHKVxM4iNP1_j1I4j6dtc8ZUy-c2FV0-J1Cnqv8Q5DrNOrd4mTx9XZ0SQr-TYJWKPGmxwsr_S-b4Z2GRbq5a0fNUo5hNTXZFqmEVy6ha78BTwp5w0wWLiKA_9-O9gLJxCVSHcrFjroGL7tYXsEXVDdlhVwLe4qcqyu2aCNqBc4DajKUSPp4iBcMYfC3JTl_DYgLTu6jfjx2ZFp_O1awjRTYjQhloqrbYpWsJlqbPYguELh41nk45GgRpfN2EqGOBygLabB3GtAe2mveCYwtQ2Kju7xIu-NPLe1UDvJHUJmDIf3JP720WB6tdNQcucgdk-CIJbiDaRUHzucNPh29deHVGb0gWsF_4xpWYuB4lnwveNCepkD5r3BQeXoMoNTrHF3ozdBjqj-93sU_lqg_etjOsn7GlWRvvbI1-7tTDP041_EIoB3oGae7acCHfmvzqbDqjOXT_zKu5RwKKJb7HuTgGaBgthcLnFlYmvMQAnuWL5OfiZuC7F_jKVVdaIYkSxdaWGUenJeUYArNPx0UYJe4xr1vi0Da48gW-rrTUjX5o0YxQO0NSuXDUXV9T1mIu0WfFQ6pOip61rDMgge8wZ9222XatRJJBDXKliaXyr1bKKxsDnz44ratcQSVlJH-OYeCzPBQkZvLzgirU6v8FDMvvV22oHgC91tbmznv78bcxL0xA9iKy9jq3OwcWnjHAbORpFjlueYSfFs0HGNxxmwvt0AWxDghrhDazZKVcK_CEeyr5FUqE41spiRAEEqClcpXt4p4EjNzFA-D_aEhZEK4Z0XVOH5f_NzTxDL3pjw0IS1-v-XTszGEtAG6zNRagJpOHPSC3bsyYSTPkoZ3G2-mYak0Mpq_nmgF1Tpy3NZ_TJVjEHIQ5N91ppDHxMcCTIkUfUUW4R98uli9zE3cIly1Lh6ggNtjYt8BKRRx8_mL0fHwa0f8_B0H9qKOMCirEw7dwLkyZoEL1TKTnwow6qFuVXbtQ3M2-nIZarxSeU7H3RCo_Pjs9g_COM-MRcYfuTUcoWAZh6ULIU3P8zuXIyOZI0TPBin5Qgk7ZE1qIWQOQBzgpZEQh88jaE5RJwtVmpc_MmKjoSeAEqaJae3LE6GzO4Mwl5p8qtZ225y2VpMAgAGu6-X-X_15vjJQ)
* [Chrome Scrolling](https://microsoft.github.io/monaco-editor/playground.html?source=v0.40.0-dev.20230621#XQAAAAJiFgAAAAAAAABBqQkHQ5NjdM0b2-jY7SIQ9S7DNlzs5W-mwj0fe1ZCDRFc9ws9XQE0SJE1jc2Vhu4mzIDyzq0Mcjp2sHIoP0RIujWPHi8B2IG4fZn4lWB1k8nG_wxyB-Maregzpyoqu8LykKh5oDCZw4ZZPyp7lquRonagZ1aKuVWkIIWPIuFXL9XeNyBlBZ8UrRrroHR4xt8uHJ44Q4QpjNYAOoEvPDvRNO8yV-YG7549C6WC6r8nNhD75FQS1yzKMpZuist1JFmXQ5rtD6Ndj2YBOMmVSD_ijeHstrBIt9PMAr0rxlrHivjkusmHDaJvWE0SQUkZKCZpEeNPatXn03hEMoHYS4y-OXAJXcbwRga9VaQrKuW-A4NvyIdgpZeAj-XU93ud7P2IXHYbUhU7eVaygey3igUqvsYOGBb-COciovDAza8Uc4jFxQCN2K6IQAjOOF2Zb739Ad1mOss_VS_-JRkRyqKfSgiApPCB-KB7Vj6xqRutkXfKUt4ZBM6P6CdqEYrd2qTR0TNkQoovTVJ7tHK8jY09FdErBAmbPokwUQ5Kd8fw6FZk1aWkQ9Ct6tghg1DfBUGXZelG2wM7LWFj3kyWeptScz01B0zUE56qVNpARvn8ZQ3f4c0OhP_NrKuDBOm1bZ6r8JlvfHYZhlvVv2HjOpEj47IDfaCoRbKozELbV080fm8grRE1RPG0FKfKcXiJ_fgQAQBRZQ4uZsDdy0qPcl_nUOyUtDijo0nDJq9GKKQJfbqL61yxAnPnlnYdsciu6OE_U-43wU3eYIfMpZeuk35bt5y7mayimqOAvRHiVJ_dA4Z5o7SVsqM6UO0vxwqktDsSnXfjB2-l4yfvO6GROmGzm_2ROouRZc4GWzK_azV44b9R_al_985RhePERlGK2z9sWuThUg-pqnMJPXIfCL2CYXOGHKVxM4iNP1_j1I4j6dtc8ZUy-c2FV0-J1Cnqv8Q5DrNOrd4mTx9XZ0SQr-TYJWKPGmxwsr_S-b4Z2GRbq5a0fNUo5hNTXZFqmEVy6ha78BTwp5w0wWLiKA_9-O9gLJxCVSHcrFjroGL7tYXsEXVDdlhVwLe4qcqyu2aCNqBc4DajKUSPp4iBcMYfC3JTl_DYgLTu6jfjx2ZFp_O1awjRTYjQhloqrbYpWsJlqbPYguELh41nk45GgRpfN2EqGOBygLabB3GtAe2mveCYwtQ2Kju7xIu-NPLe1UDvJHUJmDIf3JP720WB6tdNQcucgdk-CIJbiDaRUHzucNPh29deHVGb0gWsF_4xpWYuB4lnwveNCepkD5r3BQeXoMoNTrHF3ozdBjqj-93sU_lqg_etjOsn7GlWRvvbI1-7tTDP041_EIoB3oGae7acCHfmvzqbDqjOXT_zKu5RwKKJb7HuTgGaBgthcLnFlYmvMQAnuWL5OfiZuC7F_jKVVdaIYkSxdaWGUenJeUYArNPx0UYJe4xr1vi0Da48gW-rrTUjX5o0YxQO0NSuXDUXV9T1mIu0WfFQ6pOip61rDMgge8wZ9222XatRJJBDXKliaXyr1bKKxsDnz44ratcQSVlJH-OYeCzPBQkZvLzgirU6v8FDMvvV22oHgC91tbmznv78bcxL0xA9iKy9jq3OwcWnjHAbORpFjlueYSfFs0HGNxxmwvt0AWxDghrhDazZKVcK_CEeyr5FUqE41spiRAEEqClcpXt4p4EjNzFA-D_aEhZEK4Z0XVOH5f_NzTxDL3pjw0IS1-v-XTszGEtAG6zNRagJpOHPSC3bsyYSTPkoZ3G2-mYak0Mpq_nmgF1Tpy3NZ_TJVjEHIQ5N91ppDHxMcCTIkUfUUW4R98uli9zE3cIly1Lh6ggNtjYt8BKRRx8_mL0fHwa0f8_B0H9qKOMCirEw7dwLkyZoEL1TKTnwow6qFuVXbtQ3M2-nIZarxSeU7H3RCo_Pjs9g_COM-MRcYfuTUcoWAZh6ULIU3P8zuXIyXV-aBfrTZiLRTrjMFGxyYFIXb_h2m9aSc_lZ58r4V2P5jiEqS0FH4LVuOlJwT7kzskzdMIFZk8PwhugPUInppBCT7PUo-pjGmlB3_7yhiN8)

Smooth scrolling feels a bit different in these examples. Personally, I prefer the smooth scrolling of chrome, but others prefer the smooth scrolling of the monaco editor.

**Comment by hediet** [comment]
[And if you don't like smooth scrolling at all, please upvote this issue](https://microsoft.github.io/monaco-editor/playground.html?source=v0.40.0-dev.20230621#XQAAAAI8FgAAAAAAAABBqQkHQ5NjdM0b2-jY7SIQ9S7DNlzs5W-mwj0fe1ZCDRFc9ws9XQE0SJE1jc2Vhu4mzIDyzq0Mcjp2sHIoP0QuwA8JzV8B2IG4fZn4lWB1k8nG_wxyB-Maregzpyoqu8LykKh5oDCZw4ZZPyp7lquRonagZ1aKuVWkIIWPIuFXL9XeNyBlBZ8UrRrroHR4xt8uHJ44Q4QpjNYAOoEvPDvRNO8yV-YG7549C6WC6r8nNhD75FQS1yzKMpZuist1JFmXQ5rtD6Ndj2YBOMmVSD_ijeHstrBIt9PMAr0rxlrHivjkusmHDaJvWE0SQUkZKCZpEeNPatXn03hEMoHYS4y-OXAJXcbwRga9VaQrKuW-A4NvyIdgpZeAj-XU93ud7P2IXHYbUhU7eVaygey3igUqvsYOGBb-COciovDAza8Uc4jFxQCN2K6IQAjOOF2Zb739Ad1mOss_VS_-JRkRyqKfSgiApPCB-KB7Vj6xqRutkXfKUt4ZBM6P6CdqEYrd2qTR0TNkQoovTVJ7tHK8jY09FdErBAmbPokwUQ5Kd8fw6FZk1aWkQ9Ct6tghg1DfBUGXZelG2wM7LWFj3kyWeptScz01B0zUE56qVNpARvn8ZQ3f4c0OhP_NrKuDBOm1bZ6r8JlvfHYZhlvVv2HjOpEj47IDfaCoRbKozELbV080fm8grRE1RPG0FKfKcXiJ_fgQAQBRZQ4uZsDdy0qPcl_nUOyUtDijo0nDJq9GKKQJfbqL61yxAnPnlnYdsciu6OE_U-43wU3eYIfMpZeuk35bt5y7mayimqOAvRHiVJ_dA4Z5o7SVsqM6UO0vxwqktDsSnXfjB2-l4yfvO6GROmGzm_2ROouRZc4GWzK_azV44b9R_al_985RhePERlGK2z9sWuThUg-pqnMJPXIfCL2CYXOGHKVxM4iNP1_j1I4j6dtc8ZUy-c2FV0-J1Cnqv8Q5DrNOrd4mTx9XZ0SQr-TYJWKPGmxwsr_S-b4Z2GRbq5a0fNUo5hNTXZFqmEVy6ha78BTwp5w0wWLiKA_9-O9gLJxCVSHcrFjroGL7tYXsEXVDdlhVwLe4qcqyu2aCNqBc4DajKUSPp4iBcMYfC3JTl_DYgLTu6jfjx2ZFp_O1awjRTYjQhloqrbYpWsJlqbPYguELh41nk45GgRpfN2EqGOBygLabB3GtAe2mveCYwtQ2Kju7xIu-NPLe1UDvJHUJmDIf3JP720WB6tdNQcucgdk-CIJbiDaRUHzucNPh29deHVGb0gWsF_4xpWYuB4lnwveNCepkD5r3BQeXoMoNTrHF3ozdBjqj-93sU_lqg_etjOsn7GlWRvvbI1-7tTDP041_EIoB3oGae7acCHfmvzqbDqjOXT_zKu5RwKKJb7HuTgGaBgthcLnFlYmvMQAnuWL5OfiZuC7F_jKVVdaIYkSxdaWGUenJeUYArNPx0UYJe4xr1vi0Da48gW-rrTUjX5o0YxQO0NSuXDUXV9T1mIu0WfFQ6pOip61rDMgge8wZ9222XatRJJBDXKliaXyr1bKKxsDnz44ratcQSVlJH-OYeCzPBQkZvLzgirU6v8FDMvvV22oHgC91tbmznv78bcxL0xA9iKy9jq3OwcWnjHAbORpFjlueYSfFs0HGNxxmwvt0AWxDghrhDazZKVcK_CEeyr5FUqE41spiRAEEqClcpXt4p4EjNzFA-D_aEhZEK4Z0XVOH5f_NzTxDL3pjw0IS1-v-XTszGEtAG6zNRagJpOHPSC3bsyYSTPkoZ3G2-mYak0Mpq_nmgF1Tpy3NZ_TJVjEHIQ5N91ppDHxMcCTIkUfUUW4R98uli9zE3cIly1Lh6ggNtjYt8BKRRx8_mL0fHwa0f8_B0H9qKOMCirEw7dwLkyZoEL1TKTnwow6qFuVXbtQ3M2-nIZarxSeU7H3RCo_Pjs9g_COM-MRcYfuTUcoWAZh6ULIU3P8zuXIyNrKCXxiRCiJxW6uX8wNa_lJyogI5VLgY9xIHnOkJgW03Y3ntkjLmjbLU1pIQlYh3WFnqA6Vq9svjSgYlqf_q6AnW)

**Comment by niclasmattsson** [comment]
The Chrome one was marginally better - but neither of these are smooth-scrolling. There is a LOT of confusion about what people mean, so let me be clear. There are three levels of scrolling smoothness:

1. No smooth scroll: for every notch of scrolled on the mousewheel, the editor window instantly jumps n lines at once to a new position (n = 3 is a common default in Windows I believe).
2. Fake smooth scroll: same fundamental behavior as point 1, but with smoothly animated *transitions* between the two positions. But crucially there is no control of the transition, and no way to end up anywhere in-between the two positions. This is the best you can do with an ordinary notched mouse wheel, and this is what various VS Code settings (incorrectly IMO) call "smooth scroll".
3. Real smooth scroll: what you get from a trackpad or by dragging the scrollbar - full control of every pixel of the transition, whether fast or slow. These days when more or more mouse wheels have a free spin mode, there is no reason why you couldn't get this from a (free spin) mouse wheel too. Except VS Code hasn't implemented it yet.

I have smooth scroll enabled for my Logitech MX Anywhere 2S free-spin mouse. It works perfectly in Chrome and drawing/photo-editing applications but VS Code just won't do it. So it's super annoying when a VS Code developer refers to "smooth-scrolling like Chrome" and shows a demo with fake smooth scroll. Which triggered me to write this lengthy comment, sorry.

**Comment by hediet** [comment]
@niclasmattsson good point, I was not aware of this. I have the Logitech MX 2 with free scrolling support, do you now how/where I can try this "real" smooth scrolling on windows?

**Comment by niclasmattsson** [comment]
Yes - install Logitech Options+, then just enable smooth scrolling for the wheel. Not sure if there are other ways for Logitech mice but this works for me. I've heard that Microsoft Precision mice do this by default - that may have been the entire selling point.

https://www.logitech.com/en-us/software/logi-options-plus.html

**Comment by hediet** [comment]
> Yes - install Logitech Options+, then just enable smooth scrolling for the wheel.

Just tried it and it is nice in free scrolling mode (also works nicely with the monaco editor), but in ratchet mode it scrolls awfully slow everywhere, making the feature unusable for me.
